content/en/docs/concepts/cloud-provider/ccm-webhook.md

```diff
@@ -0,0 +1,140 @@
+        builder.RegisterWebhook("example-webhook", app.WebhookConfig{
+                Path:             examplewebhook.Path,
+                AdmissionHandler: examplewebhook.ValidatingWebhook,
+        })
+        builder.SetStopChannel(wait.NeverStop)
+        builder.SetCloudInitializer(cloudInitializer)
+        builder.SetProviderDefaults(cpoptions.ProviderDefaults{
+                WebhookBindPort: &examplewebhook.Webhookport,
+        })
+        command := builder.BuildCommand()
+
+        if err := command.Execute(); err != nil {
+                klog.Fatalf("unable to execute command: %v", err)
+        }
+}
+```
+
+Finally, a webhook configuration needs to be created, referencing the webhook endpoint which was just created.
+
+```yaml
+// Todo a webhook configuration yaml
+---
+apiVersion: admissionregistration.k8s.io/v1
+kind: ValidatingWebhookConfiguration
+metadata:
+  name: "ccm-webhook.example.com"
+webhooks:
+  - name: "ccm-webhook.example.com"
+    clientConfig:
+      service:
+        namespace: "example-namespace"
+        name: "example-service"
+    caBundle: <CA_BUNDLE>
+    rules:
+      - operations: ["CREATE"]
+        apiGroups: [""]
+        apiVersions: ["v1"]
+        resources: ["pods"]
+    namespaceSelector:
+      matchLabels:
+        modify: "true"
+    sideEffects: None
+    timeoutSeconds: 5
+    admissionReviewVersions: ["v1"]
+```
+
+{{< note >}}
+You must replace the `<CA_BUNDLE>` in the above example by a valid CA bundle
+which is a PEM-encoded CA bundle for validating the webhook's server certificate.
+{{< /note >}}
+
+The `scope` field specifies if only cluster-scoped resources ("Cluster") or namespace-scoped
+resources ("Namespaced") will match this rule. "&lowast;" means that there are no scope restrictions.
+
+{{< note >}}
+When using `clientConfig.service`, the server cert must be valid for
+`<svc_name>.<svc_namespace>.svc`.
+{{< /note >}}
+
+{{< note >}}
+Default timeout for a webhook call is 10 seconds,
+You can set the `timeout` and it is encouraged to use a short timeout for webhooks.
+If the webhook call times out, the request is handled according to the webhook's
+failure policy.
+{{< /note >}}
+
+For more information about webhook configuration, refer to [the dynamic admission control documentation](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/).
+
+## Persistent Volume Labeller
+
+One application of the cloud controller manager webhook framework is an easy
+mechanism for cloud providers to host a webhook for replacing the [persistent
+volume labelling admission
+controller](https://github.com/kubernetes/kubernetes/blob/master/plugin/pkg/admission/storage/persistentvolume/label/admission.go).
+This admission controller is responsible for adding labels to PersistentVolume
+CREATE requests for manually provisioned persistent volumes.
+
+## {{% heading "whatsnext" %}}
```

Comment:
need to clean this up?

Reply:
I copied this from another page... I was expecting it to render like the what's next section in https://kubernetes.io/docs/concepts/security/pod-security-admission/